app/forms/declaration/DeclarationAdditionalActors.scala

```diff
@@ -17,23 +17,28 @@
 package forms.declaration
 
 import forms.DeclarationPage
+import forms.Mapping.requiredRadio
 import forms.common.Eori
-import forms.declaration.DeclarationAdditionalActors.PartyType
-import play.api.data.Forms.{optional, text}
-import play.api.data.{Form, Forms}
-import play.api.libs.json.{Format, JsValue, Json, OFormat, Reads}
+import play.api.data.{Form, Forms, Mapping}
+import play.api.libs.json.{Format, JsValue, Json}
 import uk.gov.voa.play.form.ConditionalMappings._
 import utils.validators.forms.FieldValidator._
 
 case class DeclarationAdditionalActors(eori: Option[Eori], partyType: Option[String]) {
 
+  import DeclarationAdditionalActors._
+
   def isDefined: Boolean = eori.isDefined && partyType.isDefined
 
-  def toJson: JsValue = Json.toJson(this)(DeclarationAdditionalActors.format)
+  def isAllowed: Boolean = partyType.exists(allowedPartyTypes.contains)
+
+  def toJson: JsValue = Json.toJson(this)(format)
 }
 
 object DeclarationAdditionalActors extends DeclarationPage {
 
+  val maxNumberOfActors = 99
```

Comment:
To me this field should be a responsibility of `DeclarationAdditionalActorsData` class.

Reply:
Good idea.